### ISSUE TITLE
Truffle fails while processing JSON output because of the missing `deployedGeneratedSources.ast.nodes` key

### ISSUE THREAD

**Issue by cameel** [trufflesuite/truffle#3419]
Originally reported in Solidity bug tracker as https://github.com/ethereum/solidity/issues/9980.

## Issue

While running tests Truffle expects to find a `nodes` key inside `deployedGeneratedSources.ast` in the JSON output of the compiler. If it's not there or is not iterable, it fails.

The problem occurs with solc 0.7.2 and 0.7.3. `deployedGeneratedSources` is a new feature added in 0.7.2 (see https://github.com/ethereum/solidity/issues/8807 / https://github.com/ethereum/solidity/issues/9053).

## Steps to Reproduce
Just run this in the shell in an empty directory:
```bash
truffle init

cat << EOF > test/All.js
contract('All', function(accounts) {
  it('Test');
});
EOF

cat << EOF > contracts/Test.sol
pragma experimental ABIEncoderV2;

contract Test {
    function foo(uint[1] memory) public {}
}
EOF

cat << EOF > truffle-config.js
module.exports = {
  compilers: {
    solc: {
      version: "0.7.3"
    }
  }
}
EOF

truffle test
```

## Expected Behavior

```
Using network 'test'.


Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.



  Contract: All
    - Test


  0 passing (19ms)
  1 pending
```

## Actual Results

```
Compiling your contracts...
===========================
> Compiling ./contracts/Migrations.sol
> Compiling ./contracts/Test.sol
> Compilation warnings encountered:

    /tmp/a/contracts/Test.sol: Warning: SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.
,/tmp/a/contracts/Test.sol: Warning: Source file does not specify required compiler version! Consider adding "pragma solidity ^0.7.3;"

> Artifacts written to /tmp/test--144923-q2gafy2pQgb0
> Compiled successfully using:
   - solc: 0.7.3+commit.9bfce1f6.Emscripten.clang



  Contract: All
    1) "before all" hook: prepare suite for "Test"


  0 passing (17ms)
  1 failing

  1) Contract: All
       "before all" hook: prepare suite for "Test":
     TypeError: ast.nodes is not iterable
      at WireDecoder.collectUserDefinedTypes (/usr/lib/node_modules/truffle/build/webpack:/packages/decoder/dist/decoders.js:130:1)
      at new WireDecoder (/usr/lib/node_modules/truffle/build/webpack:/packages/decoder/dist/decoders.js:101:10)
      at Object.<anonymous> (/usr/lib/node_modules/truffle/build/webpack:/packages/decoder/dist/index.js:187:1)
      at Generator.next (<anonymous>)
      at /usr/lib/node_modules/truffle/build/webpack:/packages/decoder/dist/index.js:145:1
      at new Promise (<anonymous>)
      at __awaiter (/usr/lib/node_modules/truffle/build/webpack:/packages/decoder/dist/index.js:141:1)
      at Object.forProject (/usr/lib/node_modules/truffle/build/webpack:/packages/decoder/dist/index.js:185:1)
      at TestRunner.initialize (/usr/lib/node_modules/truffle/build/webpack:/packages/core/lib/testing/TestRunner.js:67:1)
      at process._tickCallback (internal/process/next_tick.js:68:7)
```

## Environment

* Operating System: Arch Linux
* Ethereum client: -
* Truffle version: v5.1.47
* node version: v10.22.1
* npm version 6.14.7

**Comment by haltman-at** [comment]
Thanks for filing this!  Fortunately this is an easy fix so I've put up a PR.

**Comment by gnidan** [comment]
Those are some grade A steps to reproduce @cameel, that's awesome! Thank you!

**Comment by cameel** [comment]
Yeah, I know how useful a good repro can be :) Thanks for handling it quickly!

**Comment by adridadou** [comment]
Is this fix already released ? if not, is there a way to fix that in the meantime?

**Comment by cameel** [comment]
Github shows that the commit (21c6b89010394c512bbe16f54d7a83a5aca03675) has been included in 5.1.48 so looks like it's already released

**Comment by haltman-at** [comment]
Yes, this fix went out in Thursday's release.
